rest/slurmrestapi.py

```diff
@@ -118,20 +119,38 @@ def login():
 @cache()
 def get_jobs():
     jobs = pyslurm.job().get()
+    onlyUsersJobs = False
+    currentUser = None
+    result = {}
+    if auth_enabled:
+        # Fetch the attributs of private_data
+        config = pyslurm.config().get()
+        private_data = config["private_data_list"]
+        # Retrive the role of current user
+        currentUser = get_current_user()
+        if private_data and 'jobs' in private_data and currentUser.role != 'admin':
+            onlyUsersJobs = True
 
     for jobid, job in jobs.iteritems():
         # add login and user's name (additionally to UID) for each job
         try:
             fill_job_user(job)
         except (KeyError):
             pass
+        # Add if jobs' owner is the current user
+        if onlyUsersJobs and job['login'] == currentUser.login:
+            result[jobid] = job
 
         # convert nodeset in array of nodes
         if job["nodes"] is not None:
             jobs[jobid]["nodeset"] = list(
                 NodeSet(job["nodes"].encode('ascii', 'ignore'))
             )
-    return jobs
+
+    if onlyUsersJobs:
```

Comment:
This if should be removed. We should have only one dict of jobs at this point, whatever the value of onlyUsersJobs is true or false.